src/App.tsx

```diff
@@ -1,26 +1,244 @@
+import { ErrorMessage } from './components/ErrorMessage';
+import { TempTodo } from './components/TempTodo';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todosFromServer, setTodosFromServer] = useState<TodoType[]>([]);
+  const [filterStatus, setFilterStatus] = useState<Filter>(Filter.All);
+  const [inputValue, setInputValue] = useState('');
+  const [tempTodo, setTempTodo] = useState<null | string>(null);
+  const [lockInput, setLockInput] = useState(false);
+  const [deleteAll, setDeleteAll] = useState(false);
+
+  const [loadingError, setLoadingError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const [todoAddError, setTodoAddError] = useState(false);
+  const [deleteError, setDeleteError] = useState(false);
+
+  const inputElement = useRef<HTMLInputElement>(null);
+
+  const loadTodos = async () => {
+    try {
+      const todos = await getTodos();
+
+      setTodosFromServer(todos);
+    } catch {
+      setLoadingError(true);
+    }
+  };
+
+  useEffect(() => {
+    if (inputElement.current) {
+      inputElement.current.focus();
+    }
+
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    if (inputElement.current) {
+      inputElement.current.focus();
+    }
+  }, [lockInput]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setInputValue(value.trim());
+  };
+
+  const handleDeleteAllErrors = useCallback(() => {
+    setLoadingError(false);
+    setTitleError(false);
+    setTodoAddError(false);
+    setDeleteError(false);
+  }, []);
+
+  const handleDeleteTodo = useCallback(
+    async (id: number) => {
+      handleDeleteAllErrors();
+      setLockInput(true);
+      try {
+        await deleteTodo(id);
+
+        setTodosFromServer(
+          [...todosFromServer].filter(todoItem => todoItem.id !== id),
+        );
+      } catch {
+        setDeleteError(true);
+      } finally {
+        setLockInput(false);
+      }
+    },
+    [handleDeleteAllErrors, todosFromServer],
+  );
+
+  const handlePostTodo = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!inputValue.length) {
+      setTitleError(true);
+
+      return;
+    }
+
+    handleDeleteAllErrors();
+
+    setLockInput(true);
+    setTempTodo(inputValue);
+    try {
+      const newPost = await postTodo(inputValue);
+
+      setTodosFromServer(currentTodos => [...currentTodos, newPost]);
+
+      setInputValue('');
+    } catch {
+      setTodoAddError(true);
+    } finally {
+      setLockInput(false);
+      setTempTodo(null);
+    }
+  };
+
+  const handleDeleteAll = async () => {
+    setDeleteAll(true);
+
+    try {
+      const deleteAllActive = todosFromServer.map(async todo => {
+        const { id, completed } = todo;
+
+        if (completed) {
+          await handleDeleteTodo(id);
+        }
+      });
+
+      await Promise.all(deleteAllActive);
```

Comment:
test will expect handling each separate delete request. so better in forEach loop send delete request.
in success cases todo should be deleted.

Reply:
I didn't fully understand how to use forEach since I need to pass an array into Promise.allSettled(), which doesn't work with this array iteration method. Therefore, I used the map method.